### ISSUE TITLE
Proposal: Building a Proof of Attendance Protocol (POAP) on Kaia

### ISSUE THREAD

**Issue by jorshimayor** [kaiachain/kaia-dapp-mono#280]
Description
- I'll create a Proof of Attendance Protocol (POAP) on the Kaia blockchain, leveraging the KIP-17 token standard to issue unique NFT badges for event participation. This ensures attendees of an event receive verifiable and collectible rewards for engaging in virtual or in-person events.

Bounty track
-other

**Comment by zxstim** [comment]
can you make it into something like a ticketing dapp? people can sign up for an NFT, then exchange the NFT on-site for a POAP NFT to prove they actually came.

**Comment by zxstim** [comment]
stick to erc721 (please use OpenZeppelin or Solady) but do explore advanced options like erc6551

**Comment by jorshimayor** [comment]
> stick to erc721 (please use OpenZeppelin or Solady) but do explore advanced options like erc6551

Okay, will do.

**Comment by zxstim** [comment]
for ticketing dapp, something simple first, think about you sending a lu.ma link, then people register and mint an NFT

then go on-site to switch out the NFT for a POAP

but you need to think about how to make this in terms of process

everything can be janky but please submit a proof of concept (frontend + smart contract)

**Comment by jorshimayor** [comment]
> can you make it into something like a ticketing dapp? people can sign up for an NFT, then exchange the NFT on-site for a POAP NFT to prove they actually came.

Alright, I'll work on that and give you feedback

**Comment by jorshimayor** [comment]
> for ticketing dapp, something simple first, think about you sending a lu.ma link, then people register and mint an NFT
> 
> then go on-site to switch out the NFT for a POAP
> 
> but you need to think about how to make this in terms of process
> 
> everything can be janky but please submit a proof of concept (frontend + smart contract)

Yes, was thinking something similar, I'll work on a simple workflow(frontend and smart contract) and see if it can be improved overtime

**Comment by jorshimayor** [comment]
Also, any reason why we're using erc721 over KIP-17?

**Comment by zxstim** [comment]
There is a community actively maintaining it and better compatibility with ethereum ecosystem

Doing kip7 or kip17 don't mean i will give you more rewards. I want to see good software aiming at end users

**Comment by jorshimayor** [comment]
Alright, make sense, I'll work on that. Thanks

**Comment by jorshimayor** [comment]
Hey,

Just made a small POC for this, I used Google calendar API instead of luma API cx it's paid, you could also upload csv file as a means to get the attendance.

Please let me know how it can improve, here's the repo:

https://github.com/jorshimayor/ticketing-dapp

Preview link: https://ticketingdapp-i6g5hhq9t-jorshimayors-projects.vercel.app

**Comment by zxstim** [comment]
can you make it public access? vercel link shows access required

**Comment by jorshimayor** [comment]
Oh okay, kindly check now, I made the demo link public, I think. Thanks

**Comment by zxstim** [comment]
<img width="950" alt="Image" src="https://github.com/user-attachments/assets/e2957cce-9819-4c50-8657-d6e023d37fd8" />

**Comment by jorshimayor** [comment]
Okay, moved it to prod: 

https://ticketingdapp.vercel.app/

Thanks

**Comment by zxstim** [comment]
i tried to mint tickets but not working (database error). can you record a demo for me?

**Comment by jorshimayor** [comment]
Here's the link to the demo video I recorded. 

Let me know what you think as it's been made public access. 

https://drive.google.com/file/d/1q_SCfyLtS6GFJb_UdwHOmD3l1wskQ_bw/view?usp=drive_link

Thanks.

**Comment by jorshimayor** [comment]
@zxstim I'd like to get your input on the PoC demo. Thanks
